syntax = "proto3";

service UserService {
    rpc Register (RegisterRequest) returns (AuthResponse);
    rpc Login (LoginRequest) returns (AuthResponse);
}

message RegisterRequest {
    string email_or_phone = 1;
    string otp = 2;
    string password = 3;
    string username = 4;
    bool use_biometric = 5;
}

message LoginRequest {
    string email_or_phone = 1;
    string password = 2;
    bool use_biometric = 3;
}

message AuthResponse {
    bool success = 1;
    string message = 2;
    string username = 3;
}
